optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/phase/event/TestdataStepScoreListener.java

```diff
@@ -0,0 +1,26 @@
+package org.optaplanner.core.impl.testdata.phase.event;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.optaplanner.core.impl.phase.event.PhaseLifecycleListenerAdapter;
+import org.optaplanner.core.impl.phase.scope.AbstractStepScope;
+import org.optaplanner.core.impl.testdata.domain.TestdataSolution;
+
+public class TestdataStepScoreListener extends PhaseLifecycleListenerAdapter<TestdataSolution> {
+
+    private List<Integer> scores = new ArrayList<>();
+
+    @Override
+    public void stepEnded(AbstractStepScope<TestdataSolution> stepScope) {
+        TestdataSolution solution = stepScope.getWorkingSolution();
+
+        if (solution.getScore() != null) {
+            scores.add(solution.getScore().getScore());
+        }
+    }
+
+    public List<Integer> getScores() {
```

Comment:
Any reason for this `Integer` extraction from `Score`? You can compare scores directly, I couldn't think of a need for this.